opm/simulators/wells/GasLiftStage2.cpp

```diff
@@ -627,8 +628,10 @@ removeSurplusALQ_(const Group &group,
         const auto well_name = dec_grad_itr->first;
         auto eco_grad = dec_grad_itr->second;
         bool remove = false;
-        if (state.checkOilTarget(eco_grad) || state.checkGasTarget()
-              || state.checkLiquidTarget(eco_grad) || state.checkWaterTarget() || state.checkALQlimit())
+        const auto delta = state.computeDelta(well_name);
+        if (state.checkOilTarget(delta[0]) || state.checkGasTarget(delta[1])
+              || state.checkLiquidTarget(delta[0] + delta[2]) || state.checkWaterTarget(delta[2])
```

Comment:
The last update looks good! A minor thing: In the above line, could we destructure the array `delta` into references like on line 1058: 
```
        const auto delta = state.computeDelta(well_name);
        auto& [delta_oil, delta_gas, delta_water, delta_alq] = delta;
        if (state.checkOilTarget(delta_oil) || state.checkGasTarget(delta_gas)
              || state.checkLiquidTarget(delta_oil + delta_gas) || state.checkWaterTarget(delta_water)
              || state.checkALQlimit(delta_alq, delta_gas))
        {
```
This could improve the readability of the last line.